### PR TITLE
Contacts activity

### PR DESCRIPTION
**Pull request by sowjanyakch**
This PR implements ContactsActivity in Jetpack Compose using coroutines.

- [X] Display user avatar and name for contacts.
- [X] Add sticky headers for contacts.
- [X] Create 1-1 room on clicking a contact.
- [X] Implement search for contacts.

Image caching for user avatars and theming will be implemented in a different PR.

### 🖼️ Screenshots
Contacts | Search | Create room for contacts
---|---|---
contacts | search_contacts | create_room_for_contacts

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not needed
- [ ] 🔖 Capability is checked or not needed 
- [ ] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [ ] 📅 Milestone is set
- [ ] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/contacts/SearchComponent.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.nextcloud.talk.R
+
+@Composable
+fun DisplaySearch(text: String, onTextChange: (String) -> Unit, contactsViewModel: ContactsViewModel) {
+    Surface(
```
Comment (rapterjet2004) [inline app/src/main/java/com/nextcloud/talk/contacts/SearchComponent.kt:37]: Theme this to match, and the exit cross should only appear once some text as been entered into the text field

**Review by rapterjet2004 — CHANGES_REQUESTED** [review]
Great Job! 👍🏾 It works on my emulator

Just need to implement
- [ ] Create new public conversation
- [ ] Proper theming

Me and @mahibi are also doing a lot of work on #3952 to get offline conversations + chatting working. I don't think there is much overlap with your PR, but just a heads up in case there are merge conflicts.

File: app/src/main/java/com/nextcloud/talk/contacts/ContactsActivityCompose.kt [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var contactsViewModel: ContactsViewModel
+
+    @SuppressLint("UnrememberedMutableState")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        NextcloudTalkApplication.sharedApplication!!.componentApplication.inject(this)
+        contactsViewModel = ViewModelProvider(this, viewModelFactory)[ContactsViewModel::class.java]
+
+        setContent {
```
[the remark this answers is not in the document]
Comment (rapterjet2004) [reply app/src/main/java/com/nextcloud/talk/contacts/ContactsActivityCompose.kt:87]: Do you have any idea how theming will work with compose? Is it a heavy refactoring?

File: app/src/main/java/com/nextcloud/talk/api/NcApiCoroutines.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import com.nextcloud.talk.models.json.autocomplete.AutocompleteOverall
+import com.nextcloud.talk.models.json.conversations.RoomOverall
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.POST
+import retrofit2.http.Query
+import retrofit2.http.QueryMap
+import retrofit2.http.Url
+
+interface NcApiCoroutines {
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/api/NcApiCoroutines.kt:19]: :+1:

**Comment by mahibi** [comment]
Great to have a kickoff with jetpack compose :rocket: 
From my side this can be merged once the remarks from Julius are solved and issues/features should be fixed/done in followup PRs.

**Comment by mahibi** [comment]
One thing i realized when testing are some bugs that existed already before this PR:
1) I have 2 accounts on sermo and with my account "Marcel" i am not able to see the account "marcel2" in the contacts. Maybe i get something wrong but i don't have an idea which could explain why it shouldn't be shown. When i search for it, it appears, but it's not in the list by default.

2) When opening a conversation and go to Conversation info to add participants, the search is broken. When i enter a name, it won't appear in the results list. Then i delete the search and try and, and then more results appear than before.
Bug report is https://github.com/nextcloud/talk-android/issues/2037

**Review by mahibi — APPROVED** [review]

**Comment by mahibi** [comment]
@sowjanyakch 
Can you rebase on master? Or as the rebase was not done regulary and may be difficult now, you can also merge master into this branch. For future branches please always use rebase and do it regulary.

please also fix warnings from  https://app.codacy.com/gh/nextcloud/talk-android/pull-requests/3936/issues

**Pull request merged**